Zend/zend_build.h

```diff
@@ -19,6 +19,12 @@
 #ifndef ZEND_BUILD_H
 #define ZEND_BUILD_H
 
+#ifdef PHP_WIN32
+#include "config.w32.h"
+#else
+#include "php_config.h"
```

Comment:
Cross-dependency like this doesn't look like a good idea to me.

Reply:
I don't understand - which cross-dependency do you mean?

Reply:
Engine code on PHP configs.

Reply:
This dependency already exists. Not including those config headers would break a lot of code.
The existing code though seems to include those headers "by accident", not by design (indirectly via other headers), which is fragile; my proposed change makes this explicit.

Reply:
I do not see any dependencies on php_config in Zend directory right now. Could you point me to the place you refer to when you say this dependency already exists?

Reply:
Proof; this is pointers to real code of the current master branch:
This is proof that `zend_build.h` requires `php_config.h` or `config.w32.h`:
https://github.com/php/php-src/blob/8d5953c002e9acccc373e96998ab62ab64990fa4/Zend/zend_build.h#L25
The `ZTS` macro is provided (or omitted) by those config headers. If `zend_build.h` happens to get included without previously including those headers, you'll get a broken PHP build. Therefore, the best way to avoid a breakage is to include the config headers from each source file which tests/uses those macros. Not doing so is fragile and buggy, something this PR aims to fix.
`zend_build.h` gets included by `zend_extensions.h` and `zend_modules.h`. Neither of those includes the config headers!
This is why it just happens to work anyway:
https://github.com/php/php-src/blob/8d5953c002e9acccc373e96998ab62ab64990fa4/TSRM/TSRM.h#L16-L21
This header gets included by various other headers, e.g.
https://github.com/php/php-src/blob/8d5953c002e9acccc373e96998ab62ab64990fa4/Zend/zend_portability.h#L47
If you believe what my PR does is add a cross-dependency, then this is also one.

Reply:
btw. I agree that a dependency between Zend and main doesn't look so good, and I'd really like to detangle that; but this is just a first preparatory step: make those dependencies explicit and stable, instead of accidental and fragile.

Reply:
@smalyshev is this resolved, i.e. do we agree, that the proposed `#include` is indeed necessary?
On https://wiki.php.net/rfc/include_cleanup you voted that include directives are *not* allowed to be cleaned up (though you did not participate in the discussion and did not explain why you think so); this particular change, however, is not a cleanup, but a fix for a bug that has not yet materialized, but will, once somebody adds new code which includes this header in a different order.